substrate/frame/beefy/src/lib.rs

```diff
@@ -740,15 +741,52 @@ pub trait WeightInfo {
 		validator_count: u32,
 		max_nominators_per_validator: u32,
 	) -> Weight;
+
+	fn set_new_genesis() -> Weight;
+}
+
+pub(crate) trait WeightInfoExt: WeightInfo {
 	fn report_double_voting(validator_count: u32, max_nominators_per_validator: u32) -> Weight {
 		Self::report_voting_equivocation(2, validator_count, max_nominators_per_validator)
 	}
-	fn report_fork_voting(validator_count: u32, max_nominators_per_validator: u32) -> Weight;
+
+	fn report_fork_voting<T: Config>(
+		validator_count: u32,
+		max_nominators_per_validator: u32,
+		ancestry_proof: &<T::AncestryHelper as AncestryHelper<HeaderFor<T>>>::Proof,
+	) -> Weight {
+		let _weight = <T::AncestryHelper as AncestryHelperWeightInfo<HeaderFor<T>>>::extract_validation_context()
+			.saturating_add(
+				<T::AncestryHelper as AncestryHelperWeightInfo<HeaderFor<T>>>::is_non_canonical(
+					ancestry_proof,
+				),
+			)
+			.saturating_add(Self::report_voting_equivocation(
+				1,
+				validator_count,
+				max_nominators_per_validator,
+			));
+
+		// TODO: return `_weight` here.
```

Comment:
@serban300 do we have any TODO issue for this?
```suggestion
		// TODO: https://github.com/paritytech/polkadot-sdk/issues/4523 - return `_weight` here.
```

Reply:
Yes, exactly, it's tracked in https://github.com/paritytech/polkadot-sdk/issues/4523 . Commited the suggestion. Thanks !